### ISSUE TITLE
Ranking/Skills System

### ISSUE THREAD

**Issue by acagliano** [commandblockguy/trek-datapack#16]
I propose adding a ranking/skill system to further increase the value of tactic and assignment in the game. The system would work this way. Players rank up through a number of objectives that give "experience".
- killing a player
- disabling/enabling core coolant
- being on an enemy core when it breaches
- etc.

The experience system would be somewhat exponential and players would level up through the following ranks:
- initiate
- cadet
- ensign
- lieutenant
- commander
- first officer
- captain
- vice admiral
- admiral

Here's where the going gets cool.. any time you advance a rank you gain a *skill point* which you can spend in some manner to acquire a skill. Skills would be:
Doctor:
      All allied players heal X hearts every X time. The higher the skill level, the more frequent.
Defender:
      *Net* shield deflection increases by 5% per level (capped at 100%)
Technician:
      Every X seconds, select one of the following actions (X decreases per level):
         Shields repair themselves by 1 damage point
         One phaser module repairs
Infiltrator:
      Player gains attack strength (higher per level) when on enemy ship.
More can be added...

So basically, for instance, a lieutenant could have the following skills:
Doctor, Lvl 2
Infiltrator, Lvl 1

This means there are this many more possibilities for how a match could go down.
This *could* also be an alternative to shield fuel which could also be made even more rare.

**Comment by Legend-of-iPhoenix** [comment]
I'll do this- I thought out how I'd implement it in detail after the match last night.

**Comment by Legend-of-iPhoenix** [comment]
Thoughts:

Instead of having both xp ranking points, just eliminate the complexity and use one unit. Since XP is a thing that exists in the game, I propose calling it just "points" or something.

I think it would be a good idea to make each successive upgrade in a path cost more, but each upgrade path is independent of the others, meaning they can have different base prices.
